### PR TITLE
Add docs for FluentBundle

### PR DESCRIPTION
**Pull request by JohnHeitmann**
Rebase of: https://github.com/projectfluent/fluent-rs/pull/73

### CODE REVIEW

File: fluent-bundle/src/bundle.rs [hunk 1]
```diff
@@ -22,21 +22,42 @@ pub struct Message {
...
+/// # `FluentBundle` Life Cycle
+///
+/// To create a bundle, call [`FluentBundle::new`] with a locale list that represents the best
+/// possible fallback chain for a given locale. The simplest case is a one-locale list.
+///
+/// Next, call [`add_resource`] one or more times, supplying translations in the FTL syntax. The
 /// `FluentBundle` instance is now ready to be used for localization.
 ///
 /// To format a translation, call `get_message` to retrieve a `fluent_bundle::bundle::Message` structure
 /// and then `format` it within the bundle.
```
Comment (zbraniecki) [inline fluent-bundle/src/bundle.rs]: That requires updating too. We don't have `get_message` anymore, and you just pass a string id to `format`/`format_message` or pass `id.attr` path to `format.

File: fluent-bundle/src/bundle.rs [hunk 2]
```diff
@@ -134,6 +249,74 @@ impl<'bundle> FluentBundle<'bundle> {
...
+    /// If no message is found at `path`, then `format` returns `None`.
+    /// 
+    /// In all other cases, `format` returns a string even if it
+    /// encountered errors. `format` uses two fallback techniques to
+    /// create the fallback string. If there are bad references in the
+    /// message, then they will be substituted with `'___'`. If there
+    /// are more extensive errors, then `format` will fall back to using
+    /// `path` itself as the formatted string. Sometimes, but not always,
+    /// these partial failures will emit extra error information in the
+    /// second term of the return tuple.
```
Comment (zbraniecki) [inline fluent-bundle/src/bundle.rs]: If we don't collect any of the errors encountered in the production of the string, it's an implementation shortcoming and I'd prefer not to document it as it indicates the design.

In 0.6 we'll be completing error reporting.

For that reason I'd prefer this sentence to say "Errors produced during message formatting are accumulated in the second tuple field of the Result::Err" or sth like this.

File: fluent-bundle/src/bundle.rs [hunk 3]
```diff
@@ -185,6 +368,43 @@ impl<'bundle> FluentBundle<'bundle> {
         None
     }
 
+    /// Formats both the message value and attributes identified by `message_id`
+    /// using `args` to provide variables.
```
Comment (zbraniecki) [inline fluent-bundle/src/bundle.rs]: I'd like to add a sentence here like : "This method is useful for cases where the customer wants to apply a compound message onto a UI element."

File: fluent-bundle/src/bundle.rs [hunk 4]
```diff
@@ -185,6 +368,43 @@ impl<'bundle> FluentBundle<'bundle> {
...
+    /// returns `None`.
+    /// 
+    /// In all other cases, `format_message` returns a `Message` even if it
+    /// encountered errors. `format_message` uses two fallback techniques to
+    /// create the fallback string. If there are bad references in the
+    /// message, then they will be substituted with `'___'`. If there
+    /// are more extensive errors, then `format_message` will fall back to using
+    /// `path` itself as the formatted string. Sometimes, but not always,
+    /// these partial failures will emit extra error information in the
+    /// second term of the return tuple.
```
Comment (zbraniecki) [inline fluent-bundle/src/bundle.rs]: Same comment about errors.

**Review by zbraniecki — CHANGES_REQUESTED** [review]
Thank you! I suggested minor improvements, and I'm happy to merge it once they're applied!

File: fluent-bundle/src/bundle.rs (hunk 2 above)
Comment (JohnHeitmann) [reply fluent-bundle/src/bundle.rs]: I reworked this pretty heavily. I tried to make parts of it blurrier (which error cases resulted in which behavior), and other parts more clear (that it's ok to ignore errors if you want).

**Review by zbraniecki — APPROVED** [review]

**Comment by zbraniecki** [comment]
Thank you!

**Pull request merged**